### ISSUE TITLE
ArgoCD 2.2.4 w/ new CVE fix does not allow for symlink'd values.yaml files

### ISSUE THREAD

**Issue by diranged** [argoproj/argo-cd#8379]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

In a few of our charts, we leverage symlinking `values.XX.yaml` to `values.staging.yaml` for example... where `XX` may be a dynamic cluster name value. As of ArgoCD 2.2.4, this fails:

```ComparisonError: rpc error: code = Unknown desc = value file 'values.dev1-test.yaml' resolved to outside repository root```

I am not sure if this feature was intended to be removed or not...

**To Reproduce**

Create a Helm chart where you have `values.something.yaml` symlinked to `values.someotherfile.yaml`.

**Expected behavior**

It should continue to work and just pick up the other values file, the same way that `helm ..` does.

**Screenshots**

![image](https://user-images.githubusercontent.com/768067/152572441-e04c65dd-828c-451c-ba00-a3a56898089c.png)

**Version**

ArgoCD 2.2.4

**Comment by jannfis** [comment]
Can you provide more information about the symlink, please (e.g. output of `ls -l` at the source)?

**Comment by jannfis** [comment]
Also, last time I've checked, `helm package` would dereference symlinks before packaging them. Do you create the Helm chart manually? Or is it an extracted Helm chart in Git?

**Comment by diranged** [comment]
```
$ ls -l
total 168
-rw-r--r--  1 diranged  staff    165 Feb  4 13:48 Chart.yaml
-rw-r--r--  1 diranged  staff    366 Feb  2 06:10 Makefile
-rw-r--r--  1 diranged  staff   5879 Feb  4 13:48 README.md
-rw-r--r--  1 diranged  staff   3398 Feb  2 06:10 README.md.gotmpl
drwxr-xr-x  3 diranged  staff     96 Feb  2 06:10 ci
drwxr-xr-x  7 diranged  staff    224 Feb  4 13:48 templates
-rw-r--r--  1 diranged  staff   2102 Feb  2 06:54 values.clusterA.yaml
lrwxr-xr-x  1 diranged  staff     16 Feb  4 13:48 values.clusterC.yaml -> values.prod.yaml
-rw-r--r--  1 diranged  staff   2325 Feb  2 06:54 values.clusterD.yaml
-rw-r--r--  1 diranged  staff   3001 Feb  4 13:48 values.clusterE.yaml
```

**Comment by diranged** [comment]
We do not use `helm package` but instead have Argo pointing to Git repos directly and pulling the charts from paths and targetRevisions.

**Comment by jannfis** [comment]
Thanks! That's indeed a bug, and is not intended to be prevented. Sorry about that.

Working on a fix.

**Comment by diranged** [comment]
> Thanks! That's indeed a bug, and is not intended to be prevented. Sorry about that.
> 
> Working on a fix.

Thanks!

**Comment by jannfis** [comment]
@diranged We just released 2.4.5 which includes the fix to the regression. Feel free to reopen this issue if the bug still persists.
